play-json/jvm/src/main/scala/play/api/libs/json/BigNumberParser.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2009-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package play.api.libs.json
+
+object BigNumberParser {
+
+  def parseBigDecimal(input: String, settings: JsonParserSettings): java.math.BigDecimal = {
+
+    // There is a limit of how large the numbers can be since parsing extremely
+    // large numbers (think thousand of digits) and operating on the parsed values
+    // can potentially cause a DDoS.
+    if (input.length > settings.bigDecimalParseSettings.digitsLimit) {
+      throw new DigitLimitException
+    }
+
+    // Must create the BigDecimal with a MathContext that is consistent with the limits used.
+    val bigDecimal = new java.math.BigDecimal(input, settings.bigDecimalParseSettings.mathContext)
+
+    // We should also avoid numbers with scale that are out of a safe limit
+    if (Math.abs(bigDecimal.scale) > settings.bigDecimalParseSettings.scaleLimit) {
+      throw new ScaleLimitException(bigDecimal.scale)
+    }
+
+    bigDecimal
+  }
+
+  def parseBigInteger(input: String, settings: JsonParserSettings): java.math.BigInteger = {
```

Comment:
Would probably merge it `EnvReads` (not used elsewhere, so not a shared/common function as far as I see) and directly return `JsResult`